ciao_contrib/_tools/specextract.py

```diff
@@ -891,15 +891,21 @@ def _check_merged_input(self,kwdict,toolname):
         check if the input file has been merged using its header keywords
         """
 
-        merge_key = [kwdict["TITLE"].lower(),
-                     kwdict["OBSERVER"].lower(),
-                     kwdict["OBJECT"].lower(),
-                     kwdict["OBS_ID"].lower()]
-
-        try:
-            merge_key.append(kwdict["DS_IDENT"].lower())
-        except KeyError:
-            pass
+        keys_to_check = ['TITLE', 'OBSERVER', 'OBJECT', 'OBS_ID', 'DS_IDENT']
+        merge_key = []
+        for mkey in keys_to_check:
+            if mkey in kwdict:
+                merge_key.append(kwdict[mkey].lower())
```

Comment:
So, I think all we care about in this routine is finding a value of `"merged". If so we can just drop the `merge_key` array and do something like (following not checked)

```
for key in keys_to_check:
    try:
        val = kwdict[key]
    except KeyError:
        continue

    if val.lower() == "merged":
        raise OSError("....")
````